Source/inv.cpp

```diff
@@ -704,6 +708,65 @@ bool AutoPlaceItemInInventory(int playerNumber, const ItemStruct &item, bool per
 	return done;
 }
 
+/**
+ * @brief Checks whether the given item can be placed on the specified player's inventory.
+ * If 'persistItem' is 'True', the item is also placed in the inventory, using an alternative
+ * algorithm that should typically provide better packing than AutoPlaceItemInInventory().
+ * @param playerNumber The player number on whose inventory will be checked.
+ * @param item The item to be checked.
+ * @param persistItem Pass 'True' to actually place the item in the inventory. The default is 'False'.
+ * @return 'True' in case the item can be placed on the player's inventory and 'False' otherwise.
+ */
+bool AutoPlaceItemInInventoryAlternate(int playerNumber, const ItemStruct &item, bool persistItem)
+{
+	InvXY itemSize = GetInventorySize(item);
+	bool done = false;

```

Comment:
`return true:` on success instead of tracking the state of done all over the function

Reply:
Does the updated version look any better to you? It doesn't return immediately but does return immediately at the end of each section now.

The only obvious way I can see to eliminate this extra `done` variable would be to replace all lines like
```
done = AutoPlaceItemInInventorySlot(...);
```
with
```
if (AutoPlaceItemInInventorySlot(...) {
    return true;
}
```
and I'm not convinced that's an improvement. I'll change it to do that if that's what you're meaning though?